src/witan/gateway/handler.clj

```diff
@@ -40,21 +33,16 @@
 (defn send-message!
   [ch m]
   (if-let [err (rs/check-message "1.0" m)]
-    (log/error "Failed to send a message - validation failed:" err)
+    (log/error "Failed to send a message - validation failed:" (str err))
     (send-edn! ch m)))
 
 (defn dispatch-event!
-  [key version params receipt]
-  (when-let [{:keys [ch]} (get @receipts receipt)]
-    (try
-      (send-message! ch {:message/type :event
-                         :event/key key
-                         :event/version version
-                         :event/params params
-                         :event/created-at (iso-dt-now)
-                         :command/receipt receipt})
-      (catch Exception e
-        (println "Failed to dispatch event:" e)))))
+  [ch receipt event]
+  (try
+    (log/debug "Dispatching event to" ch)
+    (send-message! ch event)
+    (catch Exception e
+      (log/error "Failed to dispatch event:" e))))
```

Comment:
It's the intended behaviour at the moment, yes. If 'send-message!' has thrown an exception then, for whatever reason, we're unable to put anything on Kafka